functions-continued/variables-are-references.md

```diff
@@ -0,0 +1,648 @@
+- `winners` is a variable that refers to the object `[81, 231, 12]` in memory
+
+## An Analogy: Variables and Valet Parking
+
+Variables and their values can be described like valet parking services.
+
+Stevie drives their blue car and needs to park at a hotel. They are gifted _free_ valet service (so fancy). The attendant takes the car and parks it in the parking spot numbered 151. The attendant gives a ticket to Stevie and says, "when you want your car back, turn in this special ticket, and your car will come out."
+
+When Stevie wants their car back, they need to show their special ticket.
+
+Variables are like the special ticket. The **ticket pointed to where the car was**.
+
+**The actual car is the actual value**. The actual car took up space in parking spot #151.
+
+There could be multiple identifiers for whatever car, bike, or vehicle is in parking spot #151. Maybe there's another, different ticket that points to the car in spot #151. Maybe there are hundreds of them.
+
+Variables work very similarly. Whenever a new value is stored in memory in a running program, that value occupies some space in memory. **Variables point to different spaces in memory**. When we read a variable, the computer grabs and presents the value at that spot in memory.
+
+In this world, maybe we could take our special ticket, and say that it points to the very nice red car in spot #152 instead. Maybe everyone could!
+
+Changing the parking spot that the tickets point to is **reassignment**.
+
+### More Analogies
+
+- Variables are cups with a label on them. We give each cup a name, such as `stevies_drink` or `number`. The cup can hold different things and we can replace the value it holds.
+
+- Variables are like home addresses, such as 123 Main St. The value would be the literal residence located at 123 Main St., walls and doors and all.
+
+- "_The Little Prince_ was not written by Antoine de Saint-Exupéry, but _The Little Prince_ was written by a man named Antoine de Saint-Exupéry."
+
+## Immutable vs. Mutable Data Types
+
+The different data types that we commonly work with in Python are `int`, `str`, `float`, `list`, `dict`, and `bool`.
+
+These data types can be classified as either **immutable** or **mutable** data types.
+
+_Mutability_ means an object's ability to change. An object of an _immutable_ data type _cannot be changed_. An object of a _mutable_ data type can be changed, modified, altered, and "mutated," and still be considered the same object.
+
+Some of Python's immutable data types are:
+
+- `int`
+- `float`
+- `decimal`
+- `string`
+
+Some of the mutable data types are:
+
+- `list`
+- `dict`
+
+Examples of mutation and modification include:
+
+- Appending or removing an item to a list
+- Adding or removing a key-value pair to a dictionary
+
+What does this _mean_ and how does it impact programming? One of the best ways to witness the differences is to learn how a computer manages them in memory, discussed below.
```

Comment:
I think this is a good section, but imo, I think the second example can be explained with piggy-backing off the first example? What do you think?

Also, this rewording focuses on presenting the examples as not necessarily tricky, but some examples and a callout.

```suggestion
Examples of operations that do _not_ mutate or modify include:

- Functions that take in a variable and return some new data, such as `type(hello_world)`
- Functions that do a command, but do not change the value, such as `print(hello_world)`
- Adding two numbers, resulting in a third number, such as `sum = 1 + 2`
- Incrementing a number stored in a variable, like `num += 1`

### !callout-info

## Adding Integers Doesn't Modify

When considering `sum = 1 + 2`, it may seem like we are modifying either the 1 or the 2 to turn it into a 3, which gets assigned to the variable `sum`. In reality, summing 1 and 2 changes neither 1 nor 2. Both retain their 1-ness and 2-ness, but they give us a new value, 3! Then `sum` gets a reference to 3, not the value 3 itself.

<br>

In the case `sum += 1`, by default, this line is evaluated as `sum = sum + 1`. Therefore, it works like the situation above! 1 retains its 1-ness and `sum` gets re-assigned to the new integer value.

### !end-callout

What does this _mean_ and how does it impact programming? One of the best ways to witness the differences is to learn how a computer manages them in memory, discussed below.
```

Reply:
Weird that a br is required to get the layout you want. 🤷   Is that due to special layout rules within a callout?

In the second example, it's missing an 'of' right at the start.  It should be:

In the case of `sum += 1`, by default, this line is evaluated as `sum = sum + 1`. Therefore, it works like the situation above! 1 retains its 1-ness and `sum` gets re-assigned to the new integer value.

And I'm fine with removing some of the += complexity. It seemed like an operator that could provide confusion about what really is being modified, so paring it down seems like a good idea.

I _don't_ know whether the additional function-focused examples are helpful here.  The implication of mutable and immutable values as function params are touched on later, so maybe they're not necessary?